R/mirror_data_to_redcap_project.R

```diff
@@ -0,0 +1,91 @@
+#' Mirror a dataframe to a REDCap project
+#'
+#' @param df_to_mirror Dataframe to be imported into the REDCap project.
+#' @param record_id_col Column in the dataframe that uniquely identifies each record.
+#' @param project_name REDCap project name.
+#' @param path_to_env_file Path to the ".env" credentials file.
+#' @param ... Additional parameters passed to \code{\link[REDCapR]{redcap_write}}.
+#'
+#' @return Same values as returned by \code{\link[REDCapR]{redcap_write}}.
+#' @export
+#'
+#' @seealso \code{\link{dataframe_to_redcap_dictionary}} to understand the REDCap data dictionary requirements.
+#' The dictionary must be created and uploaded to REDCap for \code{mirror_data_to_redcap_project} to function properly.
+#'
+#' @examples
+#' \dontrun{
+#' df <- data.frame(
+#'   pk_col = c("a1", "a2", "a3"),
+#'   integer_col = c(1, 2, 3),
+#'   numeric_col = 5.9,
+#'   character_col = c("a", "b", "c"),
+#'   date_col = as.Date("2011-03-07"),
+#'   date_time_col = as.POSIXct(strptime("2011-03-27 01:30:00", "%Y-%m-%d %H:%M:%S")),
+#'   email_col = c("test@example.com", "test.2@example.edu", "3test@example.net")
+#' )
+#' mirrored_data <- mirror_data_to_redcap_project(df, "pk_col", "test_project", "<env_path>.env")
+#' }
+#'
+mirror_data_to_redcap_project <- function(df_to_mirror,
+                                          record_id_col,
+                                          project_name,
+                                          path_to_env_file,
+                                          ...) {
+  if (!file.exists(path_to_env_file)) {
+    stop(sprintf("The .env file was not found at %s", path_to_env_file))
+  } else {
+    dotenv::load_dot_env(path_to_env_file)
+  }
```

Comment:
No, credentials will have to come from the environment. loading .env files from a file cannot be automated in Docker